turbo/snapshotsync/block_reader.go

```diff
@@ -139,6 +139,27 @@ func (back *RemoteBlockReader) Header(ctx context.Context, tx kv.Getter, hash co
 	}
 	return block.Header(), nil
 }
+
+func (back *RemoteBlockReader) HeaderByNumber(ctx context.Context, tx kv.Getter, blockHeight uint64) (*types.Header, error) {
+	header := rawdb.ReadHeaderByNumber(tx, blockHeight)
```

Comment:
here: 
- `tx` it's RemoteKV - means you can use rawdb to read anything from it
- `client` it's `interfaces/remote/ethbackend.proto` implementation - place for high-level funcs. There is only `Block()` method exists. rawdb.ReadHeaderByNumber it's only 2 requests to RemoteKV - no reason to add method headerByNumber to ethbackend.proto.
-  Call to client.Block() may look sub-optimal here - but RPCDaemon has BlocksLRU cache: see `BaseAPI.blockWithSenders`. Future idea - rpcdaemon has supbscription to new headers - it can warmup BlocksLRU there.
- GasPriceOracleBackend - does it need much historical blocks or not? if not - BlocksLRU must work well there.

Reply:
My advise:
- try to go higher level in code: for example `Oracle.SuggestTipCap` does call `HeaderByNumber` method but then use only hash from there. Means call `rawdb.ReadCanonicalHash` is enough. Probably same in another places. Maybe can remove `OracleBackend.HeaderByNumber` method.
- rawdb.ReadHeaderByNumber - reading only from DB, but `--syncmode=snap` store historical headers in Snapshots. Use `BaseAPI._blockReader.HeaderByNumber()` instead.

Reply:
Well before `BaseAPI` did not have the `HeaderByNumber` function, so I added a `HeaderReader` into the blockReader_ of BaseAPI

Reply:
@enriavil1 sorry, forgot about this task. Then everything looks good to me. Do you wanna merge it?

Reply:
No worries! Wouldn't we have the problem that `*RemoteBlockReader` is not suppose to use rawdb or am I understanding wrong? If we are not suppose to use rawdb to add what @AlexeyAkhunov  suggested. @AskAlexSharov

Reply:
- Yes, better use specialized grpc method - because with —suncmode=snap db will not have old blocks. 

- try to go higher level in code: for example Oracle.SuggestTipCap does call HeaderByNumber method but then use only hash from there. Means call rawdb.ReadCanonicalHash is enough. Probably same in another places. Maybe can remove OracleBackend.HeaderByNumber method.

Reply:
I am looking at it and it seems to just get the latest header so we don't actually need the readHeaderByNumber there like you say @AskAlexSharov. I will just see if this change every where is possible and get rid of readHeaderByNumber.

Reply:
It reading header, and then doesn’t use any fields from header - only it’s hash. We can read hash of any header from db by rawdb.ReadCanonicalHash